libcontainer/container_linux.go

```diff
@@ -423,6 +423,11 @@ func (c *Container) createExecFifo() error {
 	if err := unix.Mkfifo(fifoName, 0o622); err != nil {
 		return &os.PathError{Op: "mkfifo", Path: fifoName, Err: err}
 	}
+	defer func() {
+		if retErr != nil {
+			os.Remove(fifoName)
+		}
+	}()
```

Comment:
Do you want to add a unit test for this? The value is questionable, though. Maybe on refactors it will help.

I think making mkfifo fail will just be a matter of removing the permissions on the statedir